### ISSUE TITLE
Some benchmarks are too slow

### ISSUE THREAD

**Issue by msoos** [msoos/cryptominisat#481]
As per @cipherboy this is an issue here in particular: https://github.com/cipherboy/sat/tree/master/sat-competition-2018/differences/benchmarks-pi

**Comment by cipherboy** [comment]
Also [these](https://github.com/cipherboy/sat/tree/master/sat-competition-2018/orders/benchmarks-theta) for `w >= 4`. See description [here](https://github.com/cipherboy/sat/blob/master/sat-competition-2018/orders/README.md).

**Comment by msoos** [comment]
Wait. Are there solvers out there that are significantly faster, say at least 2x as fast? Because there is little point it trying to find out why if the difference is not significant :)

**Comment by cipherboy** [comment]
I guess you're right, CMS is the fastest on that later benchmark... Just surprised at the time difference between `w=2` and `w=4`:

`w=2`:
`c Total time               : 0.40        `
`c -- header says num vars:            702`
`c -- header says num clauses:        2653`
~snip~
```
c Parsing time: 0.00 s
c   type VSIDS  rest  conf freevar  IrrL  IrrB l/longC  l/allC RedL0 RedL1 RedL2  RedB l/longC  l/allC
c [distill] long cls tried: 28/28 cl-short:36 lit-r:2 T: 0.00 T-out: N T-r: 100.00%
c [distill] long useful: 36/28/28 lits-rem: 2 0-depth-assigns: 0 T: 0.00 T-out: N
c [distill] long cls tried: 1285/1088 cl-short:1839 lit-r:526 T: 0.00 T-out: N T-r: 99.98%
c [distill] long useful: 1839/1285/1313 lits-rem: 526 0-depth-assigns: 0 T: 0.00 T-out: N
c   glue     1    17  4097     700  2601    50    3.02    3.00  1347  1213   653   244    9.86    9.30
c [distill] long cls tried: 970/1968 cl-short:1268 lit-r:431 T: 0.01 T-out: N T-r: 99.95%
c [distill] long useful: 1268/970/2058 lits-rem: 431 0-depth-assigns: 0 T: 0.01 T-out: N
c   glue     1    45  8194     700  2601    50    3.02    3.00  2362  2671  1315   354   10.32    9.88
c [distill] long cls tried: 2128/3944 cl-short:2621 lit-r:1286 T: 0.03 T-out: N T-r: 99.79%
c [distill] long useful: 2621/2128/4096 lits-rem: 1286 0-depth-assigns: 0 T: 0.03 T-out: N
c ------- FINAL TOTAL SEARCH STATS ---------
```


`w=4`:
`c Total time               : 9404.80     `
`c -- header says num vars:           2602`
`c -- header says num clauses:       10103`
~snip~
```
c   luby     0 17778 3711K    1376 11977     0    3.84    3.84  245K  5959 19838    95   11.65   11.64
c   luby     0 17791 3715K    1376 11977     0    3.84    3.84  245K  6351   23K    95   11.73   11.73
c   luby     0 17810 3720K    1376 11977     0    3.84    3.84  245K  6883   26K    95   11.88   11.87
c   luby     0 17818 3724K    1376 11977     0    3.84    3.84  246K  6162 17434    95   11.61   11.61
c   luby     0 17819 3728K    1376 11977     0    3.84    3.84  246K  7106   20K    95   11.68   11.68
c   luby     0 17819 3732K    1376 11977     0    3.84    3.84  246K  6652   24K    95   11.75   11.75
c   luby     0 17820 3736K    1376 11977     0    3.84    3.84  246K  8195 13707    96   11.43   11.42
c   luby     0 17820 3740K    1376 11977     0    3.84    3.84  247K  6593 18958    96   11.60   11.60
c   luby     0 17820 3744K    1376 11977     0    3.84    3.84  247K  7388   22K    96   11.75   11.75
c [distill] long cls tried: 2125/247031 cl-short:3973 lit-r:4758 T: 4.91 T-out: N T-r: 89.14%
c [distill] long useful: 3973/2125/247122 lits-rem: 4758 0-depth-assigns: 0 T: 4.91 T-out: N
c   luby     0 17835 3748K    1376 11977     0    3.84    3.84  247K  8157   25K   187   11.79   11.79
c   luby     0 17851 3752K    1376 11977     0    3.84    3.84  247K  7903 15447   187   11.49   11.49
c   luby     0 17866 3756K    1376 11977     0    3.84    3.84  247K  8992 18148   188   11.54   11.54
c   luby     0 17882 3761K    1376 11977     0    3.84    3.84  248K  8067   22K   190   11.57   11.57
c   luby     0 17883 3765K    1376 11977     0    3.84    3.84  248K  9760 12155   191   11.25   11.25
c   luby     0 17904 3769K    1376 11977     0    3.84    3.84  248K 11137 14619   192   11.28   11.27
c   luby     0 17915 3773K    1375 11977     0    3.84    3.84  249K 10385 18709   205   11.29   11.28
c   luby     0 17936 3777K    1375 11977     0    3.84    3.84  249K 11713   21K   206   11.31   11.30
c   luby     0 17946 3781K    1374 11977     0    3.84    3.84  250K 11873 12350   222   11.05   11.04
c ------- FINAL TOTAL SEARCH STATS ---------
```

**Comment by msoos** [comment]
:) Glad CMS is faster! I am working on making the Gauss *much* faster so hopefully you'll be able to use that :)

**Comment by cipherboy** [comment]
To clarify, I believe this issue to still be open; on the former set of benchmarks there is still difference between other solvers and CMS.

I'm looking at this [instance](https://github.com/cipherboy/sat/blob/master/sat-competition-2018/differences/benchmarks-pi/cms-scheel-sha3-differences-w2-rp-i31-o35.cnf) in particular (from the first comment you made). According to my initial timings on CMS v5.0.1, this [took ~1h](https://github.com/cipherboy/sat/blob/master/sat-competition-2018/differences/args-pi/sha3-differences.args.hard_unsat#L1).

Maple SAT takes 9.8 minutes:
```
c This is COMiniSatPS.
c WARNING: for repeatability, setting FPU to use double precision
c ============================[ Problem Statistics ]=============================
c |                                                                             |
c |  Number of variables:           852                                         |
c |  Number of clauses:            2433                                         |
c |  Parse time:                   0.00 s                                       |
c |                                                                             |
c ============================[ Search Statistics ]==============================
c | Conflicts |          ORIGINAL         |          LEARNT          | Progress |
c |           |    Vars  Clauses Literals |    Limit  Clauses Lit/Cl |          |
c ===============================================================================
c ===============================================================================
c restarts              : 10609
c conflicts             : 6576513        (11176 /sec)
c decisions             : 7292476        (0.00 % random) (12392 /sec)
c propagations          : 605030390      (1028149 /sec)
c conflict literals     : 127487368      (47.18 % deleted)
c Memory used           : 37.00 MB
c CPU time              : 588.466 s

s UNSATISFIABLE
```


On CMS (2ae1177e2a9e312a6dc5c49498235229f97f5720) it takes 53.8 minutes:
```
c ------- FINAL TOTAL SEARCH STATS ---------
c UIP search time          : 3029.93     (93.84     % time)
c restarts                 : 57756       (338.82    confls per restart)
c blocked restarts         : 0           (0.00      per normal restart)
c time                     : 3029.93     
c decisions                : 22634355    (0.05      % random)
c propagations             : 695714245   
c decisions/conflicts      : 1.16        
c CONFLS stats
c conflicts                : 19569104    (6458.60   / sec)
c conflsBinIrred           : 0           (0.00      %)
c conflsBinRed             : 0           (0.00      %)
c conflsLongIrred          : 0           (0.00      %)
c conflsLongRed            : 0           (0.00      %)
c DEBUG((int)numConflicts - (int)(conflsBinIrred + conflsBinRed
c  + conflsLongIrred + conflsLongRed) = 19569104
c LEARNT stats
c units learnt             : 20          (0.00      % of conflicts)
c bins learnt              : 192         (0.00      % of conflicts)
c long learnt              : 19568896    (100.00    % of conflicts)
c otf-subs                 : 1742355     (0.09      /conflict)
c otf-subs implicit        : 236         (0.01      %)
c otf-subs long            : 1742119     (99.99     %)
c otf-subs learnt          : 1742152     (99.99     % otf subsumptions)
c otf-subs lits gained     : 1791781     (1.03      lits/otf subsume)
c guess different          : 0           (0.00      % of confl)
c cache hit re-learnt cl   : 0           (0.00      % of confl)
c red which0               : 38101       (0.19      % of confl)
c SEAMLESS HYPERBIN&TRANS-RED stats
c advProp called           : 0           
c hyper-bin add bin        : 0           (0.00      bin/call)
c trans-red rem irred bin  : 0           (0.00      bin/call)
c trans-red rem red bin    : 0           (0.00      bin/call)
c CONFL LITS stats
c orig                     : 587402694   (30.02     lit/confl)
c recurs-min effective     : 17673530    (90.31     % attempt successful)
c recurs-min lits          : 190537566   (32.44     % less overall)
c permDiff call%           : 0.06        (0.00      % attempt successful)
c permDiff lits-rem        : 0           (0.00      less lits/cl on attempts)
c further-min call%        : 0.00        (0.00      % attempt successful)
c bintri-min lits          : 0           (0.00      % less overall)
c cache-min lits           : 0           (0.00      % less overall)
c stamp-min call%          : 0.00        (0.00      % attempt successful)
c stamp-min lits           : 0           (0.00      % less overall)
c final avg                : 20.28       
c all-threads sum CPU time : 3029.93      s
c PROP stats
c Mbogo-props              : 11642.36    (3.84      / sec)
c MHyper-props             : 0.00        (0.00      / sec)
c Mprops                   : 695.71      (0.23      / sec)
c props/decision           : 0.00        
c props/conflict           : 0.00        
c ------- FINAL TOTAL SOLVING STATS END ---------
c 0-depth assigns          : 237         (111.79    % vars)
c 0-depth assigns by CNF   : 32          (15.09     % vars)
c probing time             : 0.69        (0.02      % time)
c -------- PROBE STATS ----------
c probe time               : 0.69        (939125.90 (Mega BP+HP)/s)
c called                   : 42          (0.02      s/call)
c unused Mega BP+HP        : 644319.64   (4108.38   est. secs)
c 0-depth-assigns          : 0           (0.00      % vars)
c bothsame                 : 0           (0.00      % visited)
c probed                   : 18104       (26383.02  probe/sec)
c loop iters               : 9292        (97.42     % var probed)
c failed                   : 0           (0.00      % of probes)
c visited                  : 0.02        M lits (99.36     % of available lits)
c bin add                  : 2           (0.22      % of bins)
c irred bin rem            : 0           (0.00      % of bins)
c red bin rem              : 1           (0.11      % of bins)
c time                     : 0.69        s
c CONFLS stats
c conflicts                : 0           (0.00      / sec)
c conflsBinIrred           : 0           (0.00      %)
c conflsBinRed             : 0           (0.00      %)
c conflsLongIrred          : 0           (0.00      %)
c conflsLongRed            : 0           (0.00      %)
c PROP stats
c Mbogo-props              : 107.60      (156.81    / sec)
c MHyper-props             : 0.02        (0.02      / sec)
c Mprops                   : 0.00        (0.00      / sec)
c -------- PROBE STATS END ----------
c OccSimplifier time       : 3.73        (0.12      % time)
c -------- OccSimplifier STATS ----------
c time                     : 3.73        (0.00      % var-elim)
c called                   : 42          (0.09      s per call)
c 0-depth assigns          : 0           (0.00      % vars)
c -------- OccSimplifier STATS END ----------
c SCC time                 : 0.08        (0.00      % time)
c ----- SCC STATS --------
c time                     : 0.08        (0.00      per call)
c called                   : 169         (0.43      new found per call)
c found                    : 72          (52.17     % of all found)
c bogoprops                : 8139627     % of all found
c ----- SCC STATS END --------
c --------- VAR REPLACE STATS ----------
c time                     : 3.39        (0.02      per call)
c trees' crown             : 69          (32.55     % of vars)
c 0-depth assigns          : 0           (0.00      % vars)
c lits replaced            : 38532       
c bin cls removed          : 314         
c long cls removed         : 262         
c long lits removed        : 144         
c bogoprops                : 48866787    
c --------- VAR REPLACE STATS END ----------
c vrep replace time        : 3.39        (0.10      % time)
c vrep tree roots          : 61          
c vrep trees' crown        : 69          (1.13      leafs/tree)
c distill time             : 182.33      (5.65      % time)
c -------- DISTILL STATS --------
c time                     : 182.33      (0.35      per call)
c timed out                : 0           (0.00      % of calls)
c distill/checked/potential: 295519     /153668/25561174
c lits-rem                 : 130656      
c 0-depth-assigns          : 0           (0.00      % of vars)
c -------- DISTILL STATS END --------
c strength cache-irred time: 0.75        (0.02      % time)
c strength cache-red time  : 23.37       (0.72      % time)
c -------- STRENGTHEN STATS --------
c --> cache-based on irred cls
c time                     : 0.75        (0.00      s/call)
c shrinked/tried/total     : 2          /250163/250163
c subsumed/tried/total     : 21         /250163/250163
c lits-rem                 : 2           (0.00      % of lits tried)
c called                   : 168         (0.00      % ran out of time)
c --> cache-based on red cls
c time                     : 23.37       (0.14      s/call)
c shrinked/tried/total     : 787        /1767823/11779259
c subsumed/tried/total     : 1477       /1767823/11779259
c lits-rem                 : 805         (0.01      % of lits tried)
c called                   : 168         (5959188.10 % ran out of time)
c -------- STRENGTHEN STATS END --------
c -------- IMPLICIT SUB STATS --------
c time                     : 0.54        (0.00      per call)
c timed out                : 0           (0.00      % of calls)
c rem bins                 : 293         
c rem tris                 : 0           
c -------- IMPLICIT SUB STATS END --------
c --------- Implication Cache Stats Start ----------
c lits having cache        : 1.92        % of decision lits
c num elems in cache/lit   : 112.50      extralits
c --------- Implication Cache Stats End   ----------
c Conflicts in UIP         : 19569104    (6061.03   confl/TOTAL_TIME_SEC)
c Total time               : 3228.68     
c Mem used                 : 35          MB
c Mem for longclauses      : 3           MB (9.78      %)
c Mem for watch alloc      : 17          MB (49.17     %)
c Mem for watch array      : 0           MB (0.03      %)
c Mem for assings&vardata  : 0           MB (0.08      %)
c Mem for implication cache: 0           MB (0.03      %)
c Mem for stamps           : 0           MB (0.04      %)
c Mem for search&solve     : 0           MB (0.13      %)
c Mem for renumberer       : 0           MB (0.03      %)
c Mem for component handler: 0           MB (0.00      %)
c Mem for occsimplifier    : 2           MB (6.04      %)
c Mem for xor-finder       : 0           MB (0.00      %)
c Mem for varReplacer&SCC  : 0           MB (0.06      %)
c Mem for impl subsume     : 0           MB (0.00      %)
c Mem for 3 distills       : 0           MB (0.00      %)
c Mem for prober+intree    : 0           MB (0.01      %)
c Accounted for mem (rss)  : 65.39       %
c Accounted for mem (vm)   : 42.26       %
s UNSATISFIABLE
```

And on lingeling takes longer.

**Comment by msoos** [comment]
There are some issues here:
1) Individual benchmarks are not interesting, classes of benchmarks are
2) Satisfiable problems are notoriously bad at this, there are problems that can be solved in 0.1s and timeout at 10000s with the same exact solver but different random seeds. This is not a joke. It's a well-known benchmark.
